### PR TITLE
Add workflow to build docker image to GitHub Container Registry

### PR DESCRIPTION
**Pull request by tomoasleep**
This PR adds a github workflow to build Docker image.

### CODE REVIEW

File: .github/workflows/build_image.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: docker/metadata-action@v4
+        id: meta
+        with:
+          images: increments/camo
+      - uses: docker/setup-qemu-action@v2
+      - uses: docker/setup-buildx-action@v2
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
```
Comment (mziyut) [inline .github/workflows/build_image.yml]: Environment secrets not defined yet?
Please pre-set the required information.

File: .github/workflows/build_image.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+jobs:
+  build_image:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - amd64
```
Comment (mziyut) [inline .github/workflows/build_image.yml]: nit: I would like to define the following for future multi-platform support.

```suggestion
          - linux/amd64
```

File: .github/workflows/build_image.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+      - name: Build and push
+        id: build
+        uses: docker/build-push-action@v4
+        with:
+          context: .
+          push: true
```
Comment (mziyut) [inline .github/workflows/build_image.yml:48]: imo: I do not want to push to the registry every time I create a pull request.
I suggest that you only push under certain conditions.

```suggestion
          push: ${{ github.event_name != 'pull_request' }}
```

File: .github/workflows/build_image.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+      - "v*"
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+env:
+  DOCKER_BUILDKIT: 1
```
Comment (mziyut) [inline .github/workflows/build_image.yml]: What is the purpose of defining `DOCKER_BUILDKIT`?
If your goal is to enable caching, define the option in `docker/build-push-action@v4`.

---

> Buildx will always build using the BuildKit engine and does not require DOCKER_BUILDKIT=1 environment variable for starting builds.

Ref: [docker/buildx: Docker CLI plugin for extended build capabilities with BuildKit](https://github.com/docker/buildx#building-with-buildx)

--- 

Ref: [Cache management with GitHub Actions | Docker Documentation](https://docs.docker.com/build/ci/github-actions/cache/)

File: .github/workflows/build_image.yml (hunk 3 above)
Comment (tomoasleep) [reply .github/workflows/build_image.yml:48]: I want to push image to the registry for pull requests in order to test changes in our infra. (I would like to hear your opinion)

docker/metadata-actions creates tags like `pr-2` for pull requests. (https://github.com/docker/metadata-action#basic)

File: .github/workflows/build_image.yml (hunk 4 above)
Comment (tomoasleep) [reply .github/workflows/build_image.yml]: My goal is to use buildkit, but it not to enable caching.

Recent Docker Desktop uses buildkit engine and I want to use the same engine on CI.
I think caching is unnecessary for the repository yet.

---

> Buildx will always build using the BuildKit engine and does not require DOCKER_BUILDKIT=1 environment variable for starting builds.

Thank you. I'll remove the redundunt `DOCKER_BUILDKIT: 1` :-)

File: .github/workflows/build_image.yml (hunk 1 above)
Comment (tomoasleep) [reply .github/workflows/build_image.yml]: Thank you. I'll configure these variables after https://github.com/increments/camo/pull/14#discussion_r1299527189 is resolved.

File: .github/workflows/build_image.yml (hunk 3 above)
Comment (mziyut) [reply .github/workflows/build_image.yml:48]: I'm concerned about the creation of tags such as `pr-2` on [this page](https://hub.docker.com/r/increments/camo/tags).
However, I feel that untagging it after the validation is done will not cause any problems, so I agree with your it.

**Review by mziyut — APPROVED** [review]
LGTM

**Comment by tomoasleep** [comment]
Modified this PR to push docker image to ghcr instead of dockerhub. (https://github.com/increments/camo/pull/14/commits/1de10e9f07359a4cce322f42e971ce17c8c41f90)

@mziyut Please re-review this PR 🙏

File: .github/workflows/build_image.yml (hunk 3 above)
Comment (tomoasleep) [reply .github/workflows/build_image.yml:48]: > However, I feel that untagging it after the validation is done will not cause any problems, so I agree with your it.

FYI: There are some actions to prune outdated `pr-*` tags introduced here. https://github.com/orgs/community/discussions/26267

**Review by mziyut — APPROVED** [review]
🚀

**Pull request merged**